### ISSUE TITLE
Refactor (avoid Duplication)

### ISSUE THREAD

**Issue by mikoloism** [AlirezaBabaeii/deno-e-task-manager#1]
https://github.com/AlirezaBabaeii/deno-e-task-manager/blob/990b65eee876d1072bfac7bd2804dea721906b47/src/middlewares/isAuthorized.ts#L5-L33

### REFACTORED

``` typescript
export const authourized = async (ctx: Context, next: any) => {
  try {
    const headers: Headers = ctx.request.headers;
    const authorization = headers.get("Authorization");
    if (!authorization) throw new Error('The "Authorization" Property Not Exists');

    const jwt = authorization.split(" ")[1];
    if (!jwt) throw new Error("Invalid token");

    const payload = await verify(jwt, key);
    if (!payload) throw new Error('Invlaid "payload" in token');

    ctx.response.status = 200;
    await next();
  } catch (error) {
    ctx.response.status = 401;
    ctx.response.body = {
      message: "Authorization Failed: ".concat(error.message),
    };
    return;
  }
};
```

### POINTS

These Lines are duplicated
https://github.com/AlirezaBabaeii/deno-e-task-manager/blob/990b65eee876d1072bfac7bd2804dea721906b47/src/middlewares/isAuthorized.ts#L10-L11
https://github.com/AlirezaBabaeii/deno-e-task-manager/blob/990b65eee876d1072bfac7bd2804dea721906b47/src/middlewares/isAuthorized.ts#L17
https://github.com/AlirezaBabaeii/deno-e-task-manager/blob/990b65eee876d1072bfac7bd2804dea721906b47/src/middlewares/isAuthorized.ts#L27-L30

**Comment by mikoloism** [comment]
The previous code most have side-effect.

### SIDE-EFFECT FREE

```typescript
export const authourized = async (ctx: Context, next: any) => {
  ctx.response.status = 401;

  const headers: Headers = ctx.request.headers;
  const authorization = headers.get("Authorization");
  if (!authorization) return (ctx.response.body = { message: 'The "Authorization" Property Not Exists' });

  const jwt = authorization.split(" ")[1];
  if (!jwt) return (ctx.response.body = { message: "Invalid token" });

  const payload = await verify(jwt, key);
  if (!payload) return (ctx.response.body = { message: 'Invlaid "payload" in token' });

  ctx.response.status = 200;
  await next();
};
```

**Comment by AlirezaBabaeii** [comment]
thanks for refactoring , was closed this issues in e05ddd2e2039e2aed767a2b60cc1da0b7e58c7b7 commit
